cgcnn2/data.py

```
@@ -285,113 +285,109 @@ def __getitem__(self, idx):
+        id_prop_data = [(cif_id, 0) for cif_id in id_prop_data]
+        self.id_prop_data = id_prop_data
+        random.seed(random_seed)
+        atom_init_file = os.path.join(self.root_dir, "atom_init.json")
+        assert os.path.exists(atom_init_file), "atom_init.json does not exist!"
+        self.ari = AtomCustomJSONInitializer(atom_init_file)
+        self.gdf = GaussianDistance(dmin=dmin, dmax=self.radius, step=step)
+
+    def __len__(self):
+        return len(self.id_prop_data)
+
+    @functools.lru_cache(maxsize=None)  # Cache loaded structures
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)